src/test/java/org/cactoos/scalar/ScalarOfTest.java

```diff
@@ -51,29 +53,68 @@ void worksWithCallable() {
     @Test
     void worksWithRunnable() {
         final Object obj = new Object();
+        final Object result = new Object();
+        final AtomicReference<Object> done = new AtomicReference<>();
         new Assertion<>(
-            "must hold the same value as given",
-            new ScalarOf<>(new RunnableOf(ignored -> { }, "ignored"), obj),
-            new ScalarHasValue<>(obj)
+            "Must convert Runnable into Scalar",
+            new ScalarOf<>(
+                new RunnableOf(
+                    () -> {
+                        done.set(result);
+                    }
+                ),
+                obj
+            ),
+            new MatcherOf<>(
+                scalar -> {
+                    final Object res = scalar.value();
+                    return res.equals(obj) && done.get().equals(result);
```

Comment:
@victornoel I think that in this context, it is better to use `==` because we actually get objects `result` and `obj`.